enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,340 @@
+`.spec.credentialsMode` from the CloudCredentials CR.
+
+Show that the operator is enabled for Token-based use by reading the CSV annotation provided by the operator author.
+
+Generate a manual ack for operator upgrade when cloud resources are involved. This can be determined by parsing the 
+Subscription objects. Ensures admin signs off that cloud resources are provisioned and working as intended before an 
+operator upgrade. This could be by setting the default subscription mode to Manual in the UI for operators that support 
+TAT.
+
+**Operator team / Operator SDK changes**: Follow new guidelines for allowing for the operator to work on token auth 
+enabled cluster. New guidelines would include the following to use CCO Token mode:
+
+- changing CRD to add CredentialRequest references and putting those referenced
+- CredentialRequests into a defined directory in the bundle
+- Add a bundle annotation claiming token-based authentication support
+- Add a bundle annotation indicating which permissions are required
+- add the projected ServiceAccount volume to the Deployment embedded in the CSV;
+- handle the Secret themselves (like today’s CCO Mint mode, read and use the credentials in the Secret whose name they 
+  know from the CredentialsRequest and is in the same Namespace)
+- Guidance to operator authors to structure controller code to alert on missing cloud credentials, not crash.
+
+SDK to support this new template. SDK to validate and in particular: alert on any permission changes between operator
+versions.
+
+### Workflow Description
+
+This diagram shows the steps a user will need to take in the proposed flow (specific to STS):
+
+![](STS_tomorrow.jpg)
+
+Making a layered operator ready to work on an STS enabled cluster will involve the following steps:
+
+Operator installation under the proposed system:
+1. user goes to console and starts operator install
+2. console detects token-based (STS) cluster, and token-based-supporting operator
+3. console prompts user to create roles+supply ARN-type string
+4. console creates subscription with ARN-type string embedded as a spec.config.env
+5. operator deployment is created with ARN-type string in env
+6. operator creates `CredentialsRequest` including the ARN-type string
+7. Cloud Credential Operator populates Secret based on `CredentialsRequest`
+8. operator loads Secret, makes cloud service requests
+
+
+For the operator author team:
+- Add CredentialRequests to the bundle, known location;
+- Add "role-arn"-type fields to the CredentialRequests
+- Add references to the CredentialRequests in the CRD spec;
+- Use SDK to validate bundle to catch permission changes or mis-configuration
+- Add eventing to report status on a CR to indicate lacking STS credentials for fully operational deploy or update.
+
+For the Cloud Credential Operator:
+- Add a Token mode that will watch for CredentialRequests (as per mint mode) and then resolve time-based tokens
```

Comment:
So is this a new Token mode like Mint or Manual? Does this then contradict the first half of the enhancement which talks about this being an implicit new functionality on Manual mode?

Reply:
Removed concept of another mode (Token) CCO in favor of CC detecting when time-based tokens are in use. So as you say, implicit functionality added to Manual mode. Making changes everywhere to reflect this new thinking.